src/mqt/bench/benchmark_generator.py

```diff
@@ -3,429 +3,382 @@
     from qiskit import QuantumCircuit
 
+if sys.version_info >= (3, 10, 0):  # pragma: no cover
+    from importlib import resources
+else:
+    import importlib_resources as resources
+
 from joblib import Parallel, delayed
 from mqt.bench import qiskit_helper, tket_helper, utils
-from mqt.bench.benchmarks import groundstate, hhl, pricingcall, pricingput, routing, shor, tsp
-
-benchmarks_module_paths_dict: dict[str, str] = {}
-timeout: int = 0
-
-
-def init_module_paths():
-    global benchmarks_module_paths_dict
-    benchmarks_module_paths_dict = {
-        "ae": "mqt.bench.benchmarks.ae",
-        "dj": "mqt.bench.benchmarks.dj",
-        "grover": "mqt.bench.benchmarks.grover",
-        "ghz": "mqt.bench.benchmarks.ghz",
-        "graphstate": "mqt.bench.benchmarks.graphstate",
-        "portfolioqaoa": "mqt.bench.benchmarks.qiskit_application_finance.portfolioqaoa",
-        "portfoliovqe": "mqt.bench.benchmarks.qiskit_application_finance.portfoliovqe",
-        "qaoa": "mqt.bench.benchmarks.qaoa",
-        "qft": "mqt.bench.benchmarks.qft",
-        "qftentangled": "mqt.bench.benchmarks.qftentangled",
-        "qgan": "mqt.bench.benchmarks.qiskit_application_ml.qgan",
-        "qpeexact": "mqt.bench.benchmarks.qpeexact",
-        "qpeinexact": "mqt.bench.benchmarks.qpeinexact",
-        "qwalk": "mqt.bench.benchmarks.qwalk",
-        "realamprandom": "mqt.bench.benchmarks.realamprandom",
-        "su2random": "mqt.bench.benchmarks.su2random",
-        "twolocalrandom": "mqt.bench.benchmarks.twolocalrandom",
-        "vqe": "mqt.bench.benchmarks.vqe",
-        "wstate": "mqt.bench.benchmarks.wstate",
-    }
-
-
-def create_benchmarks_from_config(cfg_path: str = "./config.json") -> bool:
-    init_module_paths()
-
-    with Path(cfg_path).open() as jsonfile:
-        cfg = json.load(jsonfile)
-        print("Read config successful")
-
-    global timeout
-    timeout = cfg["timeout"]
-
-    Path(utils.get_qasm_output_path()).mkdir(exist_ok=True, parents=True)
-
-    Parallel(n_jobs=-1, verbose=100)(delayed(generate_benchmark)(benchmark) for benchmark in cfg["benchmarks"])
-    return True
-
-
-def benchmark_generation_watcher(func, args):
-    class TimeoutException(Exception):  # Custom exception class
-        pass
+from mqt.bench.benchmarks import (
+    groundstate,
+    hhl,
+    pricingcall,
+    pricingput,
+    routing,
+    shor,
+    tsp,
+)
+
+
+class BenchmarkGenerator:
+    def __init__(self, cfg_path: str = "./config.json", qasm_output_path: str | None = None):
+        with Path(cfg_path).open() as jsonfile:
+            self.cfg = json.load(jsonfile)
+            print("Read config successful")
+        self.timeout = self.cfg["timeout"]
+        self.error_msg_outer_loop = "\n Problem occurred in outer loop: "
+        if qasm_output_path is None:
+            self.qasm_output_path = str(resources.files("mqt.benchviewer") / "static/files/qasm_output/")
+        else:
+            self.qasm_output_path = qasm_output_path
```

Comment:
Could be realized similar to the other file. In addition, platform-compatibility could be improved.
```suggestion
        self.qasm_output_path = qasm_output_path if qasm_output_path else str(resources.files("mqt.benchviewer") / "static" / "files" / "qasm_output")
```

Reply:
Implemented that.